"Hound Racing" spaceport mission (#6766)
endless-sky/endless-sky: data/human/human missions.txt

		payment 80000
		dialog "In the past few days, Pookie has barked incessantly and thrown up several times despite being fed perfectly on schedule, and the room you've been keeping her in will likely smell like dog urine for months. You are all too happy to return her to her owner, who pays you <payment>."
		log "Minor People" "Pocahontas (Pookie)" `A dog...`

mission "Pookie, Part 3"
	name "Stray Dog"
	description `Find a home for a stray dog you picked up on <origin>. She might have been from one of the Paradise Worlds.`
	source "Freedom"
	passengers 1
	to offer
		has "Pookie, Part 2: done"
	on offer
		conversation
			`The weather on <origin> is unusually rough today; the sky ripples with thunder while rain whips against locals brave or desperate enough to stay outside. Through the mist, you see a building with neon lettering saying, "East Mesa Indoor Racing Complex: Bets are open!"`
			choice
				`	(Check it out.)`

> 				`	(Stay dry in your ship.)`

Reviewer: ACTION CHOICES ARE FIRST PERSON.
```suggestion
				`	(Stay dry in my ship.)`
```
(Mind fixing this in that other mission of yours that just merged?)

> 					defer

Reviewer: Instead of immediately deferring after this choice, you should probably instead lead into a short line about you returning to your ship and then end the mission, instead of having this choice end the mission right there, since it doesn't seem to me like a choice that if I chose it would end the mission.

			`	"Our final contestant for the day is a most... unorthodox choice. The merchant who graciously donated her to us came with a most facinating story. She said that the hound comes from the Paradise Worlds, and her owner wanted the pet to have a luxury flight to her holiday home. Well, let's see if she can pay her way to victory here, folks." The audience snickers. "Now, give your best welcomes to Puny!" Puny is easily the frailest of the dogs, barely more than skin and bones. White whispers of hair clump around her head, chest, and ankles, and the bottom of her cage is wet. None of the other racers acknowledge her, but the crowd is rearing to volley a variety of curses and slurs at her, as though an actual Paradise World baron was in the cage. As you look at the poodle, you can't help but feel a strange sense of familiarity.`
			`	You feel your back get nudged, and turn to see a woman dressed in rags holding four buckets filled with credits. The names of the racers are crudely written on each of them. "Betting is 3,000 credits," she says unusually sweetly.`

> 			branch notpoor
> 				"credits" >= 3000
> 			choice
> 				`	(Bet on Bleedbug.)`
> 				`	(Bet on Doom.)`
> 				`	(Bet on Zinc.)`
> 				`	(Bet on Puny.)`
> 				`	"I'll pass."`
> 					goto nobet
> 			`	You dig through your pockets for a credit chip to toss into the bucket, until you realize that you don't have enough money to make a bet. Impatient with your bumbling, the woman turns to take money from a person next to you.`
> 				goto next
> 			label notpoor
> 			choice
> 				`	(Bet on Bleedbug.)`
> 				`	(Bet on Doom.)`
> 				`	(Bet on Zinc.)`
> 				`	(Bet on Puny.)`
> 				`	"I'll pass."`
> 					goto nobet

Reviewer: If the branch comes after the choice then we don't need a duplicate.
```suggestion
			choice
				`	(Bet on Bleedbug.)`
				`	(Bet on Doom.)`
				`	(Bet on Zinc.)`
				`	(Bet on Puny.)`
				`	"I'll pass."`
					goto nobet
			branch notpoor
				"credits" >= 3000
			`	You dig through your pockets for a credit chip to toss into the bucket, until you realize that you don't have enough money to make a bet. Impatient with your bumbling, the woman turns to take money from a person next to you.`
				goto next
			label notpoor
```

			action
				set "bet on hound racing"
				payment -3000
			`	You toss a chip into the bucket, and the woman eagerly turns towards a person next to you.`
				goto next
			label nobet
			`	The woman turns to take bets from a person next to you.`
			label next

> 			`	After several minutes pass, the siren sounds again, and the announcer speaks again. "Once again, we at the East Mesa Racing Complex give our thanks to everyone who supports our cause by participating in betting. Now, without further ado..."`

Reviewer: Repetitive "again."
```suggestion
			`	After several minutes pass, the siren sounds, and the announcer speaks again. "Once again, we at the East Mesa Racing Complex give our thanks to everyone who supports our cause by participating in betting. Now, without further ado..."`
```

			`	A red light on the ceiling turns on, and the audience shouts, "Three!" A second one lights up. "Two!" A third light, this time yellow, turns on. "One!"`

> 			`	A green light appears, and the crowd descends into chaos as the cages' doors open.`

Reviewer: ```suggestion
			`	A green light appears, and the crowd descends into chaos as the cage doors open.`
```

			`	The first to move is Bleedbug, desperate to get as far away from Doom as possible. Doom himself has his eyes set on a different target. As soon as the doors open, he rushes over the lane lines towards Zinc and bares his teeth for a bite. Anticipating this, Zinc dodges beneath the attack, and lunges in with her mouth open. Doom responds to the threat by lowering his head and bashing into her, and the two animals are quickly locked into a brutal mess of teeth and blood. The crowd goes wild at the spectacle unfolding before them as the brawl approaches your section of the fence.`
			choice
				`	(Continue watching.)`
				`	(Get away.)`
					goto leave

> 			`	Looking on, you are suddenly thrown backward when the beasts smash into the fence in front of you. Lying on the ground, you look down and see Zinc and Doom tearing into each other just beyond your feet.`

Reviewer: As Peter said, make it clear that the dogs busted through the fence so that it makes sense should you not move and die.

			choice

> 				`	(Get up and run.)`

Reviewer 2: Running sound a bit overreacting (in a room full of people), just moving back to a safer distance should be sufficient to get out of harms way.

Reviewer: With a massive wolf and dog in front of you, running doesn't actually sound like too big an overreaction to me.

Reviewer 2: > With a massive wolf and dog in front of you, running doesn't actually sound like too big an overreaction to me.

I agree, but the place was introduced like this: **"The inside of the racing complex is packed to capacity. Around you is a swarm of elbows and shoulders, jutting into you while worming your way through the crowd."**   This doesn't sound like a place where there is space to run. I do agree that "quickly increasing the distance between you and the massive beasts" is a proper reaction; it just cannot be called running when first-last is moving through a massive crowd of people.

					goto live
				`	(Lie still.)`
			`	As you lie still, Zinc is kicked on top of you, and Doom dashes in to continue the brawl. Within seconds, your skin is ripped to ribbons by two animals too deep in bloodlust to notice that another body has joined the fray. Your body bleeds out on the floor as the crowd cheers at the slaughter of a fellow human.`
				die
			label live
			`	You scramble to your feet, barely dodging Zinc as she's kicked back by Doom. As the two animals howl in pain, you push your way through the onlooking crowd, hoping to escape the danger. After getting some distance away, you turn back to see Zinc and Doom continuing to go at each other's throats.`
				goto next2
			label leave

> 			`	You turn and head away from the fence. Moments later, you hear two cries of pain accompanying the sound of wood splintering. Peeking back, you see that Zinc and Doom are fighting outside the track, with the crowd raving at the sight of violence.`

Reviewer: This line makes it much more clear that the fence was destroyed.

			choice
				`	(Get out before it gets ugly.)`
					goto nopelate
				`	(Stay for a while longer.)`
			`	Continuing to look on, you hear a loud bang to your right as the stagehand gasps in pain and collaspes. You turn to see that a mustached man next to you has his gun raised. Before you can react, you hear another gunshot and see a spurt of blood erupt from the man's temple. Almost instantly, the entire audience is up in arms as gunfire comes in from every direction. Acting instinctually, you duck down and crawl through the forest of legs towards the exit. Once there, you reach your hand up and push the door open before running out.`
				goto outside
			label nopelate
			`	As you turn around, you hear a volley of gunshots from behind, and decide to run out of the building, tackling the door open.`
				goto outside
			label nope
			`	You head towards the exit, pushing aside audience members completely captivated by the fight. Flinging the door open, you hear a gunshot from behind, and decide to run out.`
			label outside

> 			`	As you step outside, you are greeted by sheets of rain lashing you. The strength of the torrent almost causes you to fall backwards, but you manage to steady yourself. You take a glance back at the racing complex to find that the guard in the black uniform is missing. You turn back to begin the trek back to your ship, but halfway there, you feel something touch your lower leg.`

Reviewer: Repetitive "back."
```suggestion
			`	As you step outside, you are greeted by sheets of rain lashing you. The strength of the torrent almost causes you to fall backwards, but you manage to steady yourself. You take a glance back at the racing complex to find that the guard in the black uniform is missing. You turn to begin the trek back to your ship, but halfway there, you feel something touch your lower leg.`
```

				goto kindly
			label turner
			`	You turn around and come face-to-face with a dour-looking dog looking up at you with puppy-dog eyes. Inspecting it, you recognize the poodle as Puny, who must have snuck out of the complex during all the commotion.`
			label kindly
			choice
				`	(Pet her.)`
				`	(Shoo her.)`
					goto shoo
			`	You crouch down and rub the soggy tufts of hair on Puny's head. As you continue, a smile creeps onto her face, beaming in spite of the soaking rain. You think about what to do with her.`
			choice
				`	(Bring Puny aboard your ship.)`
				`	(Shoo Puny off.)`
					goto shoo
			`	You pick up Puny and carry her onboard. As you dry yourselves, you think back to what the announcer said about Puny being owned by someone from the Paradise Worlds, and wonder where you might be able to drop her off.`
			branch remember
				not "bet on hound racing"

> 			`	As you step into the cockpit, you recall the bet you put on the hound race. Looks like you won't be getting your money back.`
> 			label remember
> 			``

Reviewer: If we don't want the blank line at the end of the conversation, then we could do something like this:
```suggestion
			`	As you step into the cockpit, you recall the bet you put on the hound race. Looks like you won't be getting your money back.`
				decline
			label remember
			`	<Insert something here about being lucky you didn't spend any money on a bet or something. Anything to sound off the mission in some way similar to the above line.>`
```

				accept
			label shoo
			`	You wiggle your leg around to scare Puny off. She runs into the mist, vanishing from sight. You turn back and manage to reach your ship, changing into a dry set of clothes.`
			branch rememberdecline
				not "bet on hound racing"
			`	As you step into the cockpit, you recall the bet you put on the hound race. Looks like you won't be getting your money back.`
			label rememberdecline
			``

> 				decline

Reviewer: Would need to do the same thing here.

	invisible
	source "Freedom"
	destination "Hestia"
	to offer
		has "Pookie, Part 2: done"
	on complete
		fail "Pookie, Part 3"

mission "Pookie, Part 4"
	repeat
	landing
	source
		attributes "paradise"
		not planet "Hestia"
	to offer
		has "Pookie, Part 3: active"
	on offer
		conversation
			branch repeat
				has "Pookie, Part 4: offered"
			`After landing on <origin>, you look for a way to reach as many people on the planet as possible in order to find Puny's owner. One of the receptionists at a planetary information desk says that she can put out a notice for a lost dog on a planetwide notice board, if you're able to put in a detailed enough description of Puny. You describe the poodle's appearance to the best of your abilities, and also mention how she was traveling for a vacation when she was abducted.`
			`	When you check back with the receptionist the next morning, she tells you that nobody has approached the offer.`
				decline
			label repeat
			`You put a description of Puny on a planetwide notice board. By the time the next day arrives, you've received no offers in response.`
				decline

mission "Pookie, Part 5"
	landing
	source "Hestia"
	to offer

> 		has "Pookie, Part 3a: done"
> 	on offer

Reviewer: If we do this instead then we don't need Part 3a.
```suggestion
		has "Pookie, Part 3: active"
	on offer
		fail "Pookie, Part 3"
```

Author: The problem with doing it this way is that it's too slow: Pookie, Part 3 isn't removed from the mission list until taking off/reloading the save. Putting the fail in an on complete block makes it fail properly.

Reviewer: So #1938? I'd rather have cleaner mission structures that assume that that issue is eventually fixed that have mission structured to avoid what is essentially a visual bug.

				goto next
			label repeat
			`You approach a receptionist to put a description of Puny on a planetwide notice board.`
			label next
			`	The next day, you visit the receptionist. When she sees you, she tells you that one message came in response, saying that the dog was theirs. You copy the address provided and hail down a taxi.`
			`	The address brings you to a stark white mansion upon a cliff overlooking the ocean. A path snakes down from the entrance of the mansion to a large, gold-plated gate. You bring Puny out from the taxi and ring the doorbell next to the gate. A minute later, a servant in a prim suit approaches.`
			choice
				`	"I'm <first>, and I'm here to return a missing pet."`
				`	(Wait quietly.)`
					goto inside	
			`	The servant says, "Ah, you're the one with the dog. Come inside, mistress has been expecting you."`
				goto enter
			label inside

> 			`	When he spots you, the servant says, "Welcome. Come inside, mistress has been expecting you."`

Reviewer 2: I would expect Puny to recognize the house and to recognize Walters (and respond happily when recognizing Walters), I would even expect Puny to recognize Pemefrey (even when Pemefrey doesn't recognize Puny).

			`	By the time you reach the mansion, your skin is slightly sunburnt. The servant walks up to the oddly tall door and presses the doorbell before moving to the side, hands folded behind his back. The bell is answered by a woman with a monstrous hairdo, barely able to fit in the doorway. "Walters, who is this?" she says in a posh Paradise World accent.`
			`	"It's the captain who put out a description of Pookie, Madam Pemefrey," the servant replies.`
			`	When she hears this, Pemefrey's expression changes from annoyance to elation. "Really? Oh, I can't thank enough for this, dearie. How was it rescuing her? Did you have to barter with smugglers to track her down? Chase down kidnappers to free her? Battle with warlords for her safety?"`
			choice
				`	"It was an adventure and a half saving her."`
				`	"It wasn't anything too hard."`
					goto humble
			`	"I'm sure it was. Now, where's my darling Pookie?"`
				goto show
			label humble

> 			`	"Oh. Well, where's my darling Pookie?"`

Reviewer: If this is in fact the correct dog then as Peter mentioned, she should probably be reacting to seeing her owner. But perhaps Pemefrey could pass it off as you just having trained the dog to look happy in order to catch a quick buck.

And if this is in fact Pookie then since the player already transported Pookie before then there should probably be some sort of comment about that. "Wait, Pookie? You still owe me for those urine stains in my ship."

			`	Pemefrey snaps her head to look back at her servant. "Walters! I thought you were on my side!" She holds her hands on her chest. "It would be such a shame if you ruined our friendship like this. Why, I wouldn't be able to bear to keep you around if you betrayed me like that."`
			`	Walters shoots a desperate glance at you while stepping back, but otherwise says nothing.`
			`	"Now, where was I... oh, right. You. Out. Now." She claps twice, jumping Walters to attention, who beckons you and Puny back to the gate of the mansion.`
			`	You slam the door of the taxi shut as Walters locks the gate. The return journey is somewhat somber, with even Puny wearing a frown. When you look back up from Puny, you find yourself in a far more humble region of <origin>. The grandiose manors built into impossible locations are far distant, replaced with a small suburban town that would not look out of place in the Dirt Belt were it not for the picturesque mountain ranges surrounding it.`
			choice
				`	(Take a reststop in the town.)`
				`	(Keep on traveling to the spaceport.)`
					goto traveler

> 			`	You tell the taxi driver to pull into the town, and get off in the middle of the commercial district into a bustling crowd of people. As you try to find something to do, you eyes catch upon a storefront with a sign saying 'Antette and Co. Animal Adoption Agency.' You beckon Puny to follow and enter the adoption center.`

Reviewer: ```suggestion
			`	You tell the taxi driver to pull into the town, and get off in the middle of the commercial district into a bustling crowd of people. As you try to find something to do, your eyes catch upon a storefront with a sign saying "Antette and Co. Animal Adoption Agency." You beckon Puny to follow and enter the adoption center.`
```

				goto within
			label traveler

> 			`	You lean back in your seat and shift your eyes over to the side window. While this town is less wealthy than the neighboring estates, it is still unmistakably on a Paradise World, with the street lamps scrubbed and polished to a wasteful degree. As you pass through the commercial district, your eyes catch a sign that says 'Antette and Co. Animal Adoption Agency.' You tell the taxi driver to stop the car and step into the adoption center, leading Puny behind you.`

Reviewer: ```suggestion
			`	You lean back in your seat and shift your eyes over to the side window. While this town is less wealthy than the neighboring estates, it is still unmistakably on a Paradise World, with the street lamps scrubbed and polished to a wasteful degree. As you pass through the commercial district, your eyes catch a sign that says "Antette and Co. Animal Adoption Agency." You tell the taxi driver to stop the car and step into the adoption center, leading Puny behind you.`
```

			`	While this town is less wealthy than the neighboring estates, it is still unmistakably on a Paradise World, with the street lamps scrubbed and polished to a wasteful degree. As you pass through the commercial district, your eyes catch a sign that says "Antette and Co. Animal Adoption Agency."`
			choice
				`	(Investigate.)`
				`	(Ignore the sign.)`
					goto lazy
			label within
			`	You tell the taxi driver to stop the car and you walk into the adoption center, leading Puny behind you.`
			`	The inside of the agency is decorated with pale blue wallpaper. To either side of you are rows of plastic chairs, and in front of you is a receptionist looking down at his nails. When he notices you, he lazily says, "Welcome, welcome. What can I do for you."`
			`	You point down at Puny and say, "Could you find a home for this dog?"`
			`	The receptionist leans over the counter to look. "Wow, that's a sorry-looking puppy. What's her name?"`
			choice
				`	"It's Puny."`
					goto unfortunate
				`	"It's Pookie."`
				`	"I don't know."`
					goto figure

> 			`	"That's a nice name." The receptionist steps forward to pick up the poodle before saying, "We'll make sure Pookie gets plenty of care and love. Thank you for taking her to us."`

Reviewer 3: ```suggestion
			`	"That's a nice name." The receptionist steps forward to pick up the poodle before saying, "We'll make sure Pookie gets plenty of care and love. Thank you for bringing her to us."`
```

				goto end
			label unfortunate

> 			`	The receptionist winces. "That's... I'm sure we can come up with something better." He steps forward to pick up the poodle before saying, "We'll make sure she gets plenty of care and love. Thank you for taking her to us."`

Reviewer 3: ```suggestion
			`	The receptionist winces. "That's... I'm sure we can come up with something better." He steps forward to pick up the poodle before saying, "We'll make sure she gets plenty of care and love. Thank you for bringing her to us."`
```

				goto end
			label figure

> 			`	"Well, I'm sure we'll figure out a name for her eventually." The receptionist steps forward to pick up the poodle before saying, "We'll make sure she gets plenty of care and love. Thank you for taking her to us."`

Reviewer 3: ```suggestion
			`	"Well, I'm sure we'll figure out a name for her eventually." The receptionist steps forward to pick up the poodle before saying, "We'll make sure she gets plenty of care and love. Thank you for bringing her to us."`
```

			label end
			action
				log `Saved a dog from a hound-racing ring, and tried to give her a home. However, her alleged owner rejected her. Eventually decided to give her up for adoption.`
				log `Minor People` `Puny` `A poodle who was a former racer in a gambling ring. She might secretly be Pookie.`
			`	You exit the center and head back to your taxi for the rest of the trip back to the spaceport.`
				decline
			label lazy
			action

> 				log `Saved a dog from a hound-racing ring, and tried to give her a home. However, her alleged owner rejected her. Eventually decided to keep her as my own.`

Reviewer: ```suggestion
				log `Saved a dog from a hound-racing ring, and tried to give her a home. However, her alleged owner rejected her. Eventually decided to keep her.`
```
